extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryFrameProcessor.java

```diff
@@ -451,6 +453,14 @@ private boolean comparePartitionKeys(ResultRow row1, ResultRow row2, List<String
     int match = 0;
     for (String columnName : partitionColumnNames) {
       int i = frameReader.signature().indexOf(columnName);
+      if (ColumnType.STRING.equals(frameReader.signature().getColumnType(columnName).get()) && (row1.get(i) instanceof List || row2.get(i) instanceof List)) {
```

Comment:
this is an unfortunate place to add this check - and it would only care about partition columns
I think the other check which was added in #17002 should get triggered...

...and actually it would if there wouldn't be a `compare` in this class....this conditional will go away along with this whole method soon; so doesn't matter - its ok to have it like this until we get to that point....

Reply:
@kgyrtkirk Yeah I agree. We can keep it as it is until we get GlueingPartitioningOperator (I manually verified the MVD queries on my local branch with the GlueingPartitioningOperator changes, all of them error out properly).

The check in this PR isn't exhaustive, but it atleast works for some kind of queries. Not sure if that's preferable though.